skpro/regression/bayesian.py

```diff
@@ -0,0 +1,349 @@
+    >>> y_test_pred_proba = bayes_model.predict_proba(X_test)  # doctest: +SKIP
+    >>> y_test_pred = bayes_model.predict(X_test)  # doctest: +SKIP
+    """
+    _tags = {
+        # packaging info
+        # --------------
+        "authors": ["meraldoantonio"],  # authors, GitHub handles
+        "python_version": None,
+        "python_dependencies": ["pymc"],
+
+        # estimator tags
+        # --------------
+        "capability:multioutput": False,  # can the estimator handle multi-output data?
+        "capability:missing": True,  # can the estimator handle missing data?
+        "X_inner_mtype": "pd_DataFrame_Table",  # type seen in internal _fit, _predict
+        "y_inner_mtype": "pd_DataFrame_Table",  # type seen in internal _fit
+    }
+
+
+    def __init__(self, intercept_mu=0, intercept_sigma=10, slopes_mu=0, slopes_sigma=10, noise_sigma=10, chains=2, draws=2000):
+
+        # hyperparameters for priors
+        self.intercept_sigma = intercept_sigma
+        self.intercept_mu = intercept_mu
+        self.slopes_sigma = slopes_sigma
+        self.slopes_mu = slopes_mu
+        self.noise_sigma = noise_sigma
+        self.chains = chains
+        self.draws = draws
+
+        super().__init__()
+
+        # Assertions to check validity of input parameters
+        assert self.intercept_sigma > 0, "intercept_sigma must be positive"
+        assert self.slopes_sigma > 0, "slopes_sigma must be positive"
+        assert self.noise_sigma > 0, "noise_sigma must be positive"
+        assert isinstance(self.chains, int) and self.chains > 0, "chains must be a positive integer"
+        assert isinstance(self.draws, int) and self.draws > 0, "draws must be a positive integer"
+
+    def _fit(self, X, y):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            feature instances to fit regressor to
+        y : pandas DataFrame, must be same length as X
+            labels to fit regressor to
+
+        Returns
+        -------
+        self : reference to self
+        """
+
+        import pymc as pm
+        assert len(y.columns) == 1, "y must have only one column!"
+        self._X = X
+        self._y = y
+        self._y_vals = y.values[:,0] # we need a 1-dimensional array for compatibility with pymc
+        self._X_cols = X.columns
+        self._y_cols = y.columns
+        self._predict_done = False
+
+        with pm.Model(coords={"obs_id": X.index, "pred_id": X.columns}) as self.model:
+
+            # Mutable data containers
+            X_data = pm.Data("X", self._X, dims = ("obs_id", "pred_id"))
+            y_data = pm.Data("y", self._y_vals, dims = ("obs_id"))
+
+            # Priors for unknown model parameters
+            self.intercept = pm.Normal("intercept", mu=self.intercept_mu, sigma=self.intercept_sigma)
```

Comment:
design discussion point - is there a way to let the user specify these? Either as `pymc` or as skpro distributions. The latter would be interface consistent but would require translation between distribution objects, and `pymc` distributions.

Maybe in another PR though, this would complicate things - `pymc` distributions are nice because of their autodiff, but they seem to require a model context always and a name, so they do not exist on their own, or are not meant to.

Reply:
we could consider using `bambi`
https://github.com/bambinos/bambi
but that might lock some design decisions in that are not pleasant for sklearn users...